src/app/app.component.spec.ts

```diff
@@ -1,27 +1,49 @@
-import { TestBed, async } from '@angular/core/testing';
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { TestBed, async, ComponentFixture } from '@angular/core/testing';
+import { MatIconRegistry } from '@angular/material';
+import { AuthService } from './auth.service';
+
 import { AppComponent } from './app.component';
+
+
 describe('AppComponent', () => {
+  let fixture: ComponentFixture<AppComponent>;
+  let comp: AppComponent;
+  let AuthServiceStub: Partial<AuthService>;
+
+  AuthServiceStub = {};
+
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [
-        AppComponent
+        AppComponent,
       ],
+      schemas: [
+        CUSTOM_ELEMENTS_SCHEMA,
+      ],
+      providers: [
+        MatIconRegistry,
+        {provide: AuthService, useValue: AuthServiceStub },
+      ]
     }).compileComponents();
+
+    fixture = TestBed.createComponent(AppComponent);
+    comp = fixture.debugElement.componentInstance;
   }));
+
   it('should create the app', async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
-    const app = fixture.debugElement.componentInstance;
-    expect(app).toBeTruthy();
+    expect(comp).toBeTruthy();
   }));
-  it(`should have as title 'app'`, async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
-    const app = fixture.debugElement.componentInstance;
-    expect(app.title).toEqual('app');
+
+  it(`should have as title 'myBudget'`, async(() => {
+    expect(comp.title).toEqual('myBudget');
   }));
-  it('should render title in a h1 tag', async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
+
+  it('should render title in a span tag', async(() => {
     fixture.detectChanges();
     const compiled = fixture.debugElement.nativeElement;
-    expect(compiled.querySelector('h1').textContent).toContain('Welcome to app!');
+
+    expect(compiled.querySelector('span.title').textContent)
+      .toContain('myBudget');
```

Comment:
Stessa cosa come per il commento sopra. Onestamente eviterei di testare testi e suoi contenitori. E se decidiamo di metterlo in un <p>? Dovresti cambiare il test e, come detto nell'altro commento, anche più volte nel corso dello sviluppo.

Reply:
questo e' piu' un test d'integrazione. per il modo in cui Angular genera l'elemento HTML da il template html. Se title non esistesse o non fosse accessibile, lo script di compilazione della cartella di distribuzione fallirebbe a causa della validazione Typescript. quindi e' importante verificare che il codice sia compilabile per non avere errori in fase di deployment difficili da identificare